tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,144 @@
+from datetime import datetime
+
+from tap_tester import connections, menagerie, runner
+from base import DixaBaseTest
+from tap_tester.logger import LOGGER
+
+
+class DixaInterruptedSyncTest(DixaBaseTest):
+    """Test tap sets a bookmark and respects it for the next sync of a stream"""
+
+    def name(self):
+        return "tap_tester_dixa_bookmark_test"
+
+    def test_run(self):
+        """
+        - Verify an interrupted sync can resume based on the currently_syncing and stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the stream level bookmark 
+          are replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync. 
+          (All yet-to-be-synced streams must replicate before streams that were already synced)
+        """
+        
+        expected_streams = self.expected_streams()
+        expected_replication_keys = self.expected_replication_keys()
+        expected_replication_methods = self.expected_replication_method()
+
+        ##########################################################################
+        # First Sync
+        ##########################################################################
+        conn_id = connections.ensure_connection(self)
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # table and field selection
+        catalog_entries = [catalog for catalog in found_catalogs
+                           if catalog.get('tap_stream_id') in expected_streams]
+        
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, catalog_entries)
+
+        # Run a first sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Update State Between Syncs
+        ##########################################################################
+        
+        LOGGER.info("Current Bookmark: {}".format(first_sync_bookmarks))
+        interrupted_sync_states = { 'currently_syncing': 'messages',
+                                    'bookmarks': {  'conversations': first_sync_bookmarks.get('bookmarks', {}).get('conversations', {}),
+                                                    'messages': {list(expected_replication_keys['messages'])[0]: 1655856000000},
+                                                    'activity_logs': {list(expected_replication_keys['activity_logs'])[0]: self.start_date}}}        
```

Comment:
The key part to interruptible syncs is that the tap maintains the order of the streams across syncs. So if Stream A is complete and B is interrupted, we should expect on the resuming sync that B is replicated first and C second. The typical way ordering is ensured is by replicating the streams in alphabetical order, then shuffling based on the state. In this case I think for it to be a valid test `activity_logs` should be complete, `conversations` should be interrupted, and `messages` should not have been synced yet and wouldn't have a bookmark in state.

Reply:
I had discussed this in the team and learned that alphabetical ordering is recommended but is not a must have requirement, but we must have pre-defined order in sync operation. Tap-dixa implementation follows `conversations`, `messages` and then `activity_logs` stream sync order, based on that order we designed this test.

Reply:
Awesome 👍 the ordering makes sense then. Should `activity_logs` have a bookmark saved yet if it hasn't been synced at all? Normally I'd expect it to be empty. Does the tap save start date as a bookmark initially?

Reply:
Yes, it does. Please refer the below code.

https://github.com/singer-io/tap-dixa/blob/main/tap_dixa/streams/activitylogs.py#:~:text=start_date%20%3D%20singer,max_datetime%20%3D%20bookmark_datetime